src/hooks/useTxTracking.ts

```diff
@@ -22,16 +23,20 @@ export const useTxTracking = (): void => {
         const id = txId || txHash
 
         let origin = ''
+        let transaction_id
+
         if (id) {
           try {
             const txDetails = await getTxDetails(chainId, id)
             origin = txDetails.safeAppInfo?.url || ''
+            transaction_id = isSwapTxInfo(txDetails.txInfo) ? txDetails.txInfo.uid : undefined
```

Comment:
this is a bit confusing to me - why do we have to return transaction_id as undefined? Can't we just not trigger trackEvent later without it?

Reply:
If we set the `transaction_id` in the data layer once it will always be there for all subsequent events even if they don't set a `transaction_id` so we have to set it to undefined explicitely.

Reply:
confusing. The signature of the function looks like we should be tracking separate events. The dataLayer is an array of Events. Why would the transaction_id of event 1 in that array apply to event 10 in the array?